source/Octopus.Tentacle/Scripts/Kubernetes/KubernetesJobOutputStreamWriter.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Scripts.Kubernetes
+{
+    public class KubernetesJobOutputStreamWriter
```

Comment:
To confirm what this is trying to bring together:

1. The jobs themselves now have a bash wrapper that pipes the logs into one of two files based on stderr and stdout. To prevent interleaved log entries being created at this stage we keep the files separate on the initial write.
2. The  `RunningKubernetesJobScript` is constantly running,  and waits until effectively the task has been completed.
3. The `KubernetesJobOutputStreamWriter` meanwhile is sitting in a loop, reading the 2 generated files from step 1, and combining them into a single `IScriptLogWriter` log.
4. The various ScriptService implimentations meanwhile, are periodically called with `GetResponse` (in addition to from other calls) which read from single `IScriptLogWriter`, offsetting at each stage by whatever the `lastLogSequence `read was. Since we want to return a single stream over to server, this requirement is effectively what creates the need for the intermediate step, rather than leaving it as two separate `std*` files and avoiding having to combine them.

It would be nice if we didnt need the double read/write but I think I can see that we dont have too many other options based on how it works.

Reply:
this is correct. I'd love to be able to write directly to the `Output.log` but I was concerned about multiple writes on different threads causing issues (it was raised as a concern by @LukeButters)